### PR TITLE
[PAM][TR] PIM-4338: Display relevant thumbnails of assets

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-4338

| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | running
| Specs & Static     | Y
| Changelog updated  | Y

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php [hunk 1]
```diff
@@ -1,22 +0,0 @@
...
-use Oro\Bundle\DataGridBundle\Datagrid\Common\DatagridConfiguration;
-
-/**
- * Configurator interface
- *
- * @author    Nicolas Dupont <nicolas@akeneo.com>
- * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
- * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
- */
-interface ConfiguratorInterface
```
Comment (jjanvier) [inline src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php]: this file should be moved not deleted no ?

ok here, but we should keep this this in mind

**Comment by jjanvier** [comment]
specs and static ?

**Comment by jjanvier** [comment]
add the BC to the changelog please

**Comment by jjanvier** [comment]
GTM once fixed

File: src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php (hunk 1 above)
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php]: Actually it has been moved to Pim/Bundle/DataGridBundle/Datagrid/Configuration/ConfiguratorInterface.php but there is to much diff to be considered as the same file by Git.

File: src/Pim/Bundle/DataGridBundle/Resources/config/configurators.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 parameters:
-    pim_datagrid.datagrid.product.context_configurator.class: Pim\Bundle\DataGridBundle\Datagrid\Product\ContextConfigurator
-    pim_datagrid.datagrid.product.columns_configurator.class: Pim\Bundle\DataGridBundle\Datagrid\Product\ColumnsConfigurator
-    pim_datagrid.datagrid.product.filters_configurator.class: Pim\Bundle\DataGridBundle\Datagrid\Product\FiltersConfigurator
-    pim_datagrid.datagrid.product.sorters_configurator.class: Pim\Bundle\DataGridBundle\Datagrid\Product\SortersConfigurator
-    pim_datagrid.datagrid.product.group_columns_configurator.class: Pim\Bundle\DataGridBundle\Datagrid\Product\GroupColumnsConfigurator
-    pim_datagrid.datagrid.request_parameters_extractor.class: Pim\Bundle\DataGridBundle\Datagrid\RequestParametersExtractor
+    pim_datagrid.datagrid.product.context_configurator.class:       Pim\Bundle\DataGridBundle\Datagrid\Configuration\Product\ContextConfigurator
```
Comment (jjanvier) [inline src/Pim/Bundle/DataGridBundle/Resources/config/configurators.yml]: services names should be changed as you change the namespace

File: src/Pim/Bundle/DataGridBundle/Datagrid/Configuration/Product/ColumnsConfigurator.php [hunk 3]
```diff
@@ -191,7 +175,7 @@ protected function addColumns()
         );
 
         $this->configuration->offsetSetByPath(
-            sprintf(ContextConfigurator::SOURCE_PATH, ContextConfigurator::AVAILABLE_COLUMNS_KEY),
+            sprintf(self::SOURCE_PATH, self::AVAILABLE_COLUMNS_KEY),
```
Comment (grena) [inline src/Pim/Bundle/DataGridBundle/Datagrid/Configuration/Product/ColumnsConfigurator.php]: Maybe you could replace all those `self::` by `static::` ones, to allow overrides (integration?).
But maybe not necessary.

File: src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php (hunk 1 above)
Comment (grena) [inline src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php]: That's why a move should alway have its own commit :)
Git detects as delete/new file if it has more than 50% diff (in the same chunk) with reference file.

**Comment by grena** [comment]
GTM once fixed

File: src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php (hunk 1 above)
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Datagrid/Product/ConfiguratorInterface.php]: Right !

File: src/Pim/Bundle/DataGridBundle/Datagrid/Configuration/Product/ColumnsConfigurator.php (hunk 3 above)
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Datagrid/Configuration/Product/ColumnsConfigurator.php]: Nobody will override these constants imho

**Comment by phaseinducer** [comment]
Replaced by https://github.com/akeneo/pim-community-dev/pull/3157
